appimagetool.c

```diff
@@ -570,10 +637,22 @@ main (int argc, char *argv[])
         * should hopefully change that. */
 
         fprintf (stderr, "Generating squashfs...\n");
-        /* runtime is embedded into this executable
-        * http://stupefydeveloper.blogspot.de/2008/08/cc-embed-binary-data-into-elf.html */
-        int size = (int)((void *)&_binary_runtime_end - (void *)&_binary_runtime_start);
-        char *data = (char *)&_binary_runtime_start;
+        int size = 0;
+        char* data;
+        if (runtime_file) {
+            data = readFile(runtime_file, &size);
+            if (!data)
+                die("Unable to load provided runtime file");
+        } else {
+#ifdef HAVE_BINARY_RUNTIME
+            /* runtime is embedded into this executable
+            * http://stupefydeveloper.blogspot.de/2008/08/cc-embed-binary-data-into-elf.html */
+            size = (int)((void *)&_binary_runtime_end - (void *)&_binary_runtime_start);
+            data = (char *)&_binary_runtime_start;
+#else
+            die("No runtime file was provided");
```

Comment:
Unfortunately I can't do that, since _binary_runtime_end and _binary_runtime_start does not exist in OSX. Thus this code needs to be into preprocessor.

Reply:
I am pretty sure there is a better way to organize this code. The flow you propose isn't final, you could adapt it a bit. But this is not mandatory.

Reply:
The only thing that could be done is to put it in a separate method, so that it would be isolated - the logic there was already flawed anyway.